src/pages/event.js

```diff
@@ -65,19 +82,22 @@ export function Event() {
       <span className='power-title'>Power</span><span className='power-content'>{token.owner.tokensOwned}</span>
     </div>
   )
+
   useEffect(() => {
+    // Get new batch of tokens
     if (eventId) {
       dispatch(fetchEventPageData({ eventId, first: GRAPH_LIMIT, skip: GRAPH_LIMIT*pageIndex  }))
     }
   }, [dispatch, eventId, pageIndex])
 
   useEffect(() => {
-    let ownerIds = tokens.map(t => t.owner.id)
-    if (event && event.tokenCount > GRAPH_LIMIT && tokens && tokens.length > 0) {
-      const totalPages = Math.ceil(event.tokenCount / GRAPH_LIMIT);
-      if (pageIndex + 1 < totalPages) {
-        setPageIndex(pageIndex + 1);
+    // Call next batch of tokens (if there is more), then load the new tokens data
+    const totalPages = Math.ceil(event.tokenCount / GRAPH_LIMIT);
+    if (event && tokens && tokens.length > 0 && pageIndex < totalPages) {
```

Comment:
Move this conditional to one or more constants with an appropriate name according to your responsibility.